source/postcard-rpc/src/host_client/mod.rs

```diff
@@ -495,9 +492,32 @@ where
         T::Message: DeserializeOwned,
     {
         let cancel_fut = self.stopper.wait_stopped();
-        let operate_fut = self.subscribe_inner::<T>(depth);
+        let operate_fut = self.subscribe_inner::<T>(depth, true);
         select! {
             _ = cancel_fut => Err(IoClosed),
+            res = operate_fut => res.map_err(|_| IoClosed),
```

Comment:
This isn't quite true, you changed `subscribe_inner` to have two kinds of errors here, which means that in the future if we accidentally return `AlreadySubscribed` from `subscribe_inner` here, it'll be treated as if the i/o was closed.

This is generally the reason I suggested duplicating the `inner` function!

Reply:
This function can only return IoClosed, so I guess I can panic if it sees AlreadySubscribed, but it would be a bug either way in the inner function.